src/huggingface_hub/hf_api.py

```diff
@@ -2364,29 +2364,15 @@ def list_repo_likers(
                 `None`.
 
         Returns:
-            `List[User]`: a list of [`User`] objects.
+            `Iterable[User]`: an iterable of [`huggingface_hub.hf_api.User`] objects.
         """
 
         # Construct the API endpoint
         if repo_type is None:
             repo_type = constants.REPO_TYPE_MODEL
         path = f"{self.endpoint}/api/{repo_type}s/{repo_id}/likers"
-        headers = self._build_hf_headers(token=token)
-
-        # Make the request
-        response = get_session().get(path, headers=headers)
-        hf_raise_for_status(response)
-
-        # Parse the results into User objects
-        likers_data = response.json()
-        return [
-            User(
-                username=user_data["user"],
-                fullname=user_data["fullname"],
-                avatar_url=user_data["avatarUrl"],
-            )
-            for user_data in likers_data
-        ]
+        for liker in paginate(path, params={}, headers=self._build_hf_headers(token=token)):
+            yield User(username=liker["user"], fullname=liker["fullname"], avatar_url=liker["avatarUrl"])
```

Comment:
Note that this is unfortunately a breaking change since users cannot do `list_repo_likers(...)[0]` anymore. It might break some script in the wild. We already had a similar issue when we switched `list_models` (and datasets/spaces) to paginated results in the past. It was much more painful as it was used everywhere.

Here I'm not sure there is much adoption yet so I would be tempted to merge "as is" and break things rather than having a half-solution for months. @LysandreJik what's your take on this? 

For context, I've checked https://github.com/search?q=list_repo_likers&type=code&p=1 and didn't find much third-party integrations.

Reply:
I would tend to say that a deprecation cycle rarely hurts but if you're certain it's not being used much yet, then maybe breaking this one is fine

Reply:
The problem is that a deprecation cycle on an output type is a bit more annoying. I tried something for `list_models` in the past (https://github.com/huggingface/huggingface_hub/pull/1143) but we got quite some problems and misunderstanding from users... :confused:

Reply:
Since there is not so much adoption (compared to `list_models()` or `list_datasets()`), imo it's worth merging as is rather than handling a deprecation cycle